src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/multitable/MultitableFlowsSpec.groovy

```diff
@@ -1510,4 +1388,94 @@ mode with existing flows and hold flows of different table-mode types"() {
             it.multiTable = newValue
         }
     }
+
+    //TODO: move to PathFinder."get a potential flow on a path of 4 switches: multi and single"() or something like this
```

Comment:
PathFinder is interface, which is used for finding paths for application, not for testing. it must NOT contain any methods for finding test data.

Reply:
Here and below: obviously, I'm not going to add handy test methods to the application code. The fact that we use code from application in tests is a big problem.
First of all, we need to clone all the currently used code from application to test package (so tests become independent from the tested application).
Then we can extend current models, making then something bigger than just dataclasses. In this particular case I didn't know that there is interface named PathFinder in openkilda. I just thought, what would be readable class and method name for the code that is looking for an appropriate path, it was something like `PathFinder.getWhatINeed()`. It's not org.openkilda.java-src.floodlight...Pathfinder.getWhatINeed().
Overall, I see our tests (at the specs level) as human-readable stuff without calling NB, FL or reading kafka. It should be something like 
```
SwitchPair switchPair = SwitchPairFactory.getNeighboringSwitches()
Flow flow = FlowFactory.getFlowBetween(switchPair)
flow.getToDegradedState()
assert flow.getState() == DEGRADED
```

Reply:
If you want to add some classes only for testing - go ahead. I thought that you wanted to change application classes

Reply:
It still seems that you mean interface `org.openkilda.pce.finder.PathFinder` because the is no any other class with same name in our project. New guy will try to find a class with name `PathFinder` and will move code there to fix this TODO. 
Please name or note that you mean some test class.